rustls-platform-verifier/src/verification/android.rs

```diff
@@ -265,12 +264,17 @@ fn extract_result_info(env: &JNIEnv<'_>, result: JObject<'_>) -> (VerifierStatus
     // Extract the `String?`.
     let msg = env
         .get_field(result, "message", "Ljava/lang/String;")
-        .and_then(|m| m.l())
-        .map(|o| (!o.is_null()).then_some(o))
-        .and_then(|s| s.map(|s| JavaStr::from_env(env, s.into())).transpose())
-        .unwrap();
-
-    (status, msg.map(String::from))
+        .and_then(|m| m.l());
+    let msg: Option<String> = if let Ok(msg) = msg {
+        if let Ok(msg) = env.get_string((&msg).into()) {
+            Some(msg.into())
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+    (status, msg)
```

Comment:
Nit: maybe we could do it more like this?

```diff
--- a/rustls-platform-verifier/src/verification/android.rs
+++ b/rustls-platform-verifier/src/verification/android.rs
@@ -264,16 +264,10 @@ fn extract_result_info(
     // Extract the `String?`.
     let msg = env
         .get_field(result, "message", "Ljava/lang/String;")
-        .and_then(|m| m.l());
-    let msg: Option<String> = if let Ok(msg) = msg {
-        if let Ok(msg) = env.get_string((&msg).into()) {
-            Some(msg.into())
-        } else {
-            None
-        }
-    } else {
-        None
-    };
+        .and_then(|msg| m.l().ok())
+        .and_then(|msg| env.get_string(&msg).ok())
+        .map(String::from);
+
     (status, msg)
 }
```

Reply:
I tried to do that in the chain, but eventually the borrow checker won't let it pass.

Reply:
Oh, because we need to store some intermediate state on the stack? We can still probably use the same combinators even if we have to bind one of the intermediates?

Reply:
Yeah I got it worked out...had to use an unwrap() but it's ok since the value is guaranteed to be Some from previous step.

Reply:
Well nvm...it won't work because None does get passed through at that point. I'll change it again.